step-strikers-app/step-strikers-app/Battle Screens/BattleSelectActionViewController.swift

```diff
@@ -186,23 +186,41 @@ class BattleSelectActionViewController: UIViewController, UITableViewDataSource,
     }
     
     func createActionArray() {
+        // First: check if the fight action (possible for all characters) is possible
+        // Note: String format makes the number 2 digits
+        if(localCharacter.currWeapon.staminaCost <= localCharacter.currStamina) {
+            actions.append(Action(name: "Fight", staminaCost: "\(String(format: "%02d", localCharacter.currWeapon.staminaCost)) STA"))
+        }
+        
+        // Next: Add all other actions. Check against stamina for Attackers and against spell points for Casters
         let characterClass = localCharacter.getCharacterClass()
-        print(characterClass)
         if characterClass == "Fighter" {
-            for index in 0 ... (FighterActions.count-1) {
-                actions.append(Action(name:FighterActions[index].actionName, staminaCost: FighterActions[index].cost))
+            for index in 1 ... (FighterActions.count-1) {
```

Comment:
Starting at 1 for the index is to prevent fists from being an option twice right?

Reply:
Yup!